src/tir/transforms/arg_binder.cc

```diff
@@ -186,18 +191,8 @@ void ArgBinder::BindDLTensor(const Buffer& buffer, const PrimExpr& device_type,
   if (!(buffer->dtype == DataType::Int(1) || buffer->dtype == DataType::Int(4) ||
         buffer->dtype == DataType::UInt(4))) {
     auto type_msg = tvm::tir::StringImm(type_err_msg.str());
-    asserts_.emplace_back(AssertStmt(a_ndim == v_ndim, msg, nop));
```

Comment:
Was this just a duplicate?

Reply:
Yup.  The buffer's dimensionality is checked earlier, so this is entirely a duplicate check on the dimensionality.